platform/packages/finance/src/coin/mod.rs

```diff
@@ -129,11 +119,22 @@ where
     }
 }
 
+impl<C> Clone for Coin<C>
+where
+    C: ?Sized,
```

Comment:
I do not think the type safety is weakened - the only thing we did with the relaxation was to widener the accepted Coin<SOMETHING> definitions ... so we may want to have Coin<A_STUPID_NFT>, for example ... so I believe that was a harmless relaxation

Reply:
I understand that I introduces ambiguity here. By relaxation of the trait bounds I didn't mean returning `C: Currency`, but rather dropping `C: ?Sized`. I'm afraid that allowing unsizing may lead to something along the lines of the following:
```
(Coin::<A>::new(1) as Coin<dyn Currency>) + (Coin::<B>::new(2) as Coin<dyn Currency>)
```
Of course this is an oversimplification which most likely won't compile right now. The thing is while relaxing trait bounds allows more types to use a definition, they also convey information to the developer looking at it that it could and may be unsized, and that it is an allowed operation.

Keep in mind that some of our implementations already allow addition behind a reference, so boxing the type may actually allow for such code.

Reply:
I'll address this.
Partially, I have bought your arguments. Partially, I have realized that my initial motivation behind 'not-required-to-be-Sized` is wrong. I was thinking that since the trait Currency is a marker trait that would allow using the types without being instantiated.
Finance is an internal crate that we will not use with ref types as currencies ,,,, but nevertheless, I would vote for this change for its simpler code.